### ISSUE TITLE
[Feature request] improve `Hot Reload` feature

### ISSUE THREAD

**Issue by BachiMjavanadze** [vuejs/language-tools#4313]
At the moment, if we want to use the `hot reload` functionality, then we need to write this to the `VSCode` `settings.json`:

`"files.autoSave": "afterDelay",`

and run this command:

`npm run serve`

This functionality is well suited for those cases when we edit `html` or `css` but when we write a logic of the app, this is not the optimal option, because during the software development process there is constant and unnecessary compilation of code.

The solution to this problem could be if we write in `settings.json` this:

`"files.autoSave": "onFocusChange",`

or even this:

`"files.autoSave": "off",`

But it's a compromise, why compromise if you don't have to compromise?

There are two solutions; One is a button on the taskbar, how this is done in the `Visual Studio IDE 2022`

![image](https://github.com/vuejs/language-tools/assets/38082501/8c21bcdc-f49c-48b0-888a-5464affaa497)

and another solution is `Flutter` extension for `VSCode`:

https://github.com/vuejs/language-tools/assets/38082501/8e909c3f-3639-4d6d-b751-918a3ab7e6d2

As you can see from the video, `Flutter` developers have not disabled the coolest `VSCode` feature:

`"files.autoSave": "afterDelay"`

It would be good if we combine these two methods and do this: let's write again in `settings.json`:

`"files.autoSave": "afterDelay",`

then let's create a small menu on the `VSCode` taskbar:

![1](https://github.com/vuejs/language-tools/assets/38082501/f69556de-0c2c-4bd6-8c2f-bb2fbe2e5540)

run the command:

`npm run serve`

Then if we select `Hot Reload`, everything happens as before: the system looks at the changes in the files and compilation occurs instantly but if we select `Reaload on Click`:

![2](https://github.com/vuejs/language-tools/assets/38082501/1bdd160d-2c00-4cf2-b291-b4f2f512936a)

then compilation occurs only when the `Ctrl+S` buttons on keyboard (or triangle button on taskbar) are clicked.

So, my request is not only about `Hot Reload`  but smooth switching between `Hot Reload` and `Reload on Click`.

**Comment by so1ve** [comment]
Wrong repo?

**Comment by BachiMjavanadze** [comment]
> Wrong repo?

which is correct repo?

**Comment by so1ve** [comment]
You mean the vue extensions should integrate with the bundler and hot-reload when you change your code?

**Comment by BachiMjavanadze** [comment]
Yes, it would be better if we got this functionality out of the box.

**Comment by codymikol** [comment]
I think that's just added complexity for no real gain, the dev server can just watch the filesystem imo.

**Comment by BachiMjavanadze** [comment]
> I think that's just added complexity

come on, bro... for `Visual Studio IDE 2022` and for `Flutter` this does not add complexity, but for `Vue` it does?!

**Comment by so1ve** [comment]
Honestly this "hot reload" feature is provided by your bundler, such as vite and webpack, right? All you need is to save the file and wait for it to reload.

**Comment by BachiMjavanadze** [comment]
@so1ve bro, don't you see my Flutter vide example above? The Flutter developers do not disable coolest feature of VSCode: "files.autoSave": "afterDelay".  and Vue.js developer should disable this feature and manually save and this is not safe  and not comfortable... a suggest solution when you can switch between the to modes: existing behavior and new one.

you need to test it to understand

**Comment by RayGuo-ergou** [comment]
regardless of this setting

```
"files.autoSave": "afterDelay"
```

if you config your bundle right, the hot reload is an out of box thing.

**Comment by BachiMjavanadze** [comment]
> if you config your bundle right, the hot reload is an out of box thing.

first:

my request is not only about `Hot reload` but smooth switching between `hot reload` and `Reload on Click`.

second:

in my `vue.config.js` I have:

`module.exports = {
  devServer: {
  webSocketServer: false
  }
}`

and this does not reload browser ( I need to manually reload browser), but comipation still happens, it is compromise... Please give me your solution. hope I will get the experience that I described above.

**Comment by KazariEX** [comment]
Closed because this is not related to the responsibility of language tools. You should find the extension that integrate with your bundlers.
